src/js/components/channels/twilio-channel-content.jsx

```diff
@@ -1,27 +1,31 @@
 import React, { Component, PropTypes } from 'react';
-import { updateTwilioAttributes, resetTwilioAttributes, linkTwilioChannel, unlinkTwilioChannel, pingTwilioChannel } from '../../services/integrations-service';
-
-import { ReactTelephoneInput } from '../../lib/react-telephone-input';
-
 import { connect } from 'react-redux';
 import isMobile from 'ismobilejs';
 
+import { updateTwilioAttributes, resetTwilioAttributes, linkTwilioChannel, unlinkTwilioChannel, pingTwilioChannel } from '../../services/integrations-service';
+import { ReactTelephoneInput } from '../../lib/react-telephone-input';
+
 export class TwilioChannelContentComponent extends Component {
 
-    static contextTypes = {
-        settings: PropTypes.object,
-        ui: PropTypes.object
+    static propTypes = {
+        linkColor: PropTypes.string,
+        phoneNumber: PropTypes.string.isRequired,
+        linkState: PropTypes.oneOf(['unlinked', 'pending', 'linked']),
+        smoochId: PropTypes.string.isRequired,
+        text: PropTypes.object.isRequired,
+        channelState: PropTypes.object.isRequired
     };
 
     linkTwilioNumber = () => {
-        linkTwilioChannel(this.props.userId, {
```

Comment:
what do you mean? the component was getting `userId` from the `connect` part... but all channel page are receiving `smoochId` from the parent.